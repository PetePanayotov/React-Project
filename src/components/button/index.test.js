import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index'
import detailsPageHandlers from '../../utils/details-page-handlers';
const {like , dislike , deletCar , comment , removeComent} = detailsPageHandlers;

describe('Button Component' , () => {

    it('Should render Like button' , () => {
        const tree = renderer.create(
            <Button type="like" text="Like" handler={like}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render dislike button' , () => {
        const tree = renderer.create(
            <Button type="like" text="Don't Like" handler={dislike}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render Delete button' , () => {
        const tree = renderer.create(
            <Button type="delete" text="Delete" handler={deletCar}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render Comment button' , () => {
        const tree = renderer.create(
            <Button type="comment" text="Comment" handler={comment}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should render Remove Comment button' , () => {
        const tree = renderer.create(
            <Button type="remove" text="Remove" handler={removeComent}/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});