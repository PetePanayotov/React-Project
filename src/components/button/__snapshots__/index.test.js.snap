// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render Comment button 1`] = `
<div
  className="comment-button-div"
>
  <button
    className="comment-button"
    onClick={[Function]}
  >
    Comment
  </button>
</div>
`;

exports[`Button Component Should render Delete button 1`] = `
<div
  className="delete-button-div"
>
  <button
    className="delete-button"
  >
    Delete
  </button>
</div>
`;

exports[`Button Component Should render Like button 1`] = `
<div
  className="like-button-div"
>
  <button
    className="like-button"
    onClick={[Function]}
  >
    Like
  </button>
</div>
`;

exports[`Button Component Should render Remove Comment button 1`] = `
<div
  className="remove-button-div"
>
  <button
    className="remove-button"
  >
    Remove
  </button>
</div>
`;

exports[`Button Component Should render dislike button 1`] = `
<div
  className="like-button-div"
>
  <button
    className="like-button"
    onClick={[Function]}
  >
    Don't Like
  </button>
</div>
`;
